iceoryx2/src/port/waitset.rs

```diff
@@ -799,15 +853,17 @@ impl<Service: crate::service::Service> WaitSet<Service> {
     /// loop {
     ///     // blocks until an event arrives or TIMEOUT was reached, handles all arrived events
     ///     // and then returns.
-    ///     waitset.wait_and_process_once(on_event, TIMEOUT)?;
+    ///     waitset.wait_and_process_once_with_timeout(on_event, TIMEOUT)?;
     ///     // do some event post processing
     ///     println!("handled events");
     /// }
     ///
     /// # Ok(())
     /// # }
     /// ```
-    pub fn wait_and_process_once<F: FnMut(WaitSetAttachmentId<Service>) -> CallbackProgression>(
+    pub fn wait_and_process_once_with_timeout<
+        F: FnMut(WaitSetAttachmentId<Service>) -> CallbackProgression,
+    >(
         &self,
         mut fn_call: F,
         timeout: Duration,
```

Comment:
Food for thought. In other Rust APIs the `_suffix` parameter is the first, e.g. with `map_or`, where the `_or` parameter is the first and the `map` parameter the second.

Reply:
I would prefer to add the parameter since it is an addition to `wait_and_process_once`. Also in my head it behaves somehow like the C++ default parameter value.

@orecham you decide the tie

Reply:
Just a suggestion but I think the reason this is done is because the additional parameter is just a value and therefore a few characters but the closure can span a few lines and then the additional parameter is far away from the function name.